cli/objref.go

```diff
@@ -67,3 +69,113 @@ func parseNestedObjectID(ctx context.Context, startingDir fs.Entry, parts []stri
 
 	return e.(object.HasObjectID).ObjectID(), nil
 }
+
+func findSnapshotsByRootObjectID(ctx context.Context, rep repo.Repository, rootID object.ID) ([]*snapshot.Manifest, error) {
+	ids, err := snapshot.ListSnapshotManifests(ctx, rep, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error listing snapshot manifests")
+	}
+
+	manifests, err := snapshot.LoadSnapshots(ctx, rep, ids)
+	if err != nil {
+		return nil, errors.Wrap(err, "error loading snapshot manifests")
+	}
+
+	var result []*snapshot.Manifest
+
+	for _, m := range manifests {
+		if m.RootObjectID() == rootID {
+			result = append(result, m)
+		}
+	}
+
+	return result, nil
+}
+
+func findSnapshotByRootObjectIDOrManifestID(ctx context.Context, rep repo.Repository, rootID string) (*snapshot.Manifest, error) {
+	m, err := snapshot.LoadSnapshot(ctx, rep, manifest.ID(rootID))
+	if err == nil {
+		return m, nil
+	}
+
+	mans, err := findSnapshotsByRootObjectID(ctx, rep, object.ID(rootID))
+	if err != nil {
+		return nil, err
+	}
+
+	// no matching snapshots.
+	if len(mans) == 0 {
+		return nil, nil
+	}
+
+	// all snapshots have consistent metadata, pick any.
+	if areSnapshotsConsistent(mans) {
+		return mans[0], nil
+	}
+
+	// at this point we found multiple snapshots with the same root ID which don't agree on other
+	// metadata (the attributes, ACLs, ownership, etc. of the root)
+	if os.Getenv("KOPIA_REQUIRE_UNAMBIGUOUS_ROOT") != "" {
```

Comment:
* Should this be the default instead of "opt-in" ?
* It may make sense to explicitly pass this as a flag to the command and then as an option to the function instead of relying on relatively hidden environment variables. WDYT?

Reply:
Moved to be a CLI parameter, defaulting to false. Let me know if you have strong conviction about opt-out vs opt-in.

Reply:
The question is what default behavior is both safer and less surprising for users.
* If the default is out-out, then it could fail with an error when there are ambiguous roots and log information on either how to allow restores of ambiguous roots, and which snapshots could be used for the restore instead of a root id.
* If the default is opt-in, then a warning could be logged when there are ambiguous roots along with info on (a) how to prevent ambiguous roots from being restored and (b) the snapshots that could be used to restore the root with different permissions.